.github/actions/docker-scan/action.yml

```diff
@@ -11,42 +11,56 @@ inputs:
   token:
     description: "Token for allowing the action to post in the security tab"
     required: true
+  upload_docker_vulns:
+    description: "Flag to specify vulnerabilities upload"
+    required: false
+    default: true
 
 runs:
   using: "composite"
-  steps:
+  steps:      
+    - name: Re-tag docker image
+      run: |
+        docker tag ${{ inputs.docker_image }} ${{ inputs.sbom_name }}
+      shell: bash
+
     - name: Install Trivy
       run: |
         curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin v0.34.0
       shell: bash
 
     - name: run trivy docker
+      if: github.event_name == 'push' && format('refs/heads/{0}', github.event.repository.default_branch) == github.ref
```

Comment:
Could you switch up this to capture this condition in an environment variable and then check that var in all the subsequent `if` conditions:

```yaml
  - name: set upload sbom conditional
    run: |
      echo "upload_sbom=${{ github.event_name == 'push' && format('refs/heads/{0}', github.event.repository.default_branch) == github.ref }}" >> $GITHUB_ENV
```

You should then be able to check it like so on the next steps:

```yaml
    - name: run trivy docker
      if: env.upload_sbom == 'true'
```

Reply:
done!